r/src/main/java/org/molgenis/r/rock/RockConnection.java

```diff
@@ -163,15 +170,38 @@ private String getRSessionResourceUrl(String path) {
     return String.format("%s/r/session/%s%s", application.getUrl(), rockSessionId, path);
   }
 
-  private HttpHeaders createHeaders() {
-    return new HttpHeaders() {
-      {
-        String auth = application.getUser() + ":" + application.getPassword();
-        byte[] encodedAuth = Base64.getEncoder().encode(auth.getBytes(StandardCharsets.UTF_8));
-        String authHeader = "Basic " + new String(encodedAuth);
-        add("Authorization", authHeader);
-      }
-    };
+  private String getAuthHeader() {
+    String auth = application.getUser() + ":" + application.getPassword();
```

Comment:
password?

Reply:
This is such that a request from rock can use the password to authenticate so we know if the proces (that armadillo started via rock) has access to the data (from that user)? 
Kinda scary, is there not a toke we could use instead? Or a way to at least encrypt this? Or is the internal communication also SSL? Because untrusted R code might now abuse it? Or to go even further, we could symply use a session number to reconnect because the password doesn't add much security?